metals/src/main/scala/scala/meta/internal/metals/Doctor.scala

```diff
@@ -362,63 +438,77 @@ final class Doctor(
   private def extractTargetInfo(
       targetId: BuildTargetIdentifier
   ): List[DoctorTargetInfo] = {
-    val scalaDoctorInfo =
-      buildTargets.scalaTarget(targetId).map(extractScalaTargetInfo).toList
-    val javaDoctorInfo =
-      buildTargets.javaTarget(targetId).map(extractJavaTargetInfo).toList
-    scalaDoctorInfo ::: javaDoctorInfo
+    val javaTarget = buildTargets.javaTarget(targetId)
+    buildTargets
+      .scalaTarget(targetId)
+      .map(st => extractScalaTargetInfo(st, javaTarget))
+      .toList
   }
 
-  private def extractScalaTargetInfo(target: ScalaTarget) = {
+  private def extractScalaTargetInfo(
+      target: ScalaTarget,
+      javaTarget: Option[JavaTarget]
+  ) = {
     val scalaVersion = target.scalaVersion
-    val definition: String =
+    val interactive =
       if (mtagsResolver.isSupportedScalaVersion(scalaVersion)) {
-        Icons.unicode.check
+        DoctorStatus(Icons.unicode.check, isCorrect = true)
```

Comment:
It seems that there is only 3 instances of statuses so they might be extracted into companion object
```
object DoctorStatus {
   val check = DoctorStatus(Icons.unicode.check, true)
   val alert = DockerStatus(Icons.unicode.alert, false)
   val error = DocktorStatus(Icons.unicode.error, false)
}
```

Reply:
Good idea! Refactored and it looks much nicer.